Unhandled error crashed the server

The specific bug that triggered it has been resolved here https://github.com/screwdriver-cd/models/pull/50 , but the server should never exit like this:

160812/191904.322, [response] http://0.0.0.0:8080: post /v3/webhooks/github {} 500 (893ms)
160812/191906.047, [request,webhook-build,bb12771e794ac72679ea2538219fef33f7789432] data: Received status update to RUNNING
/usr/src/app/node_modules/vogels/lib/serializer.js:35
      return new Date(value).toISOString();
                             ^
RangeError: Invalid time value
    at Object.internals.serialize.date (/usr/src/app/node_modules/vogels/lib/serializer.js:35:30)
    at Object.internals.serializeAttribute.serializer.serializeAttribute [as serializeAttribute] (/usr/src/app/node_modules/vogels/lib/serializer.js:88:22)
    at /usr/src/app/node_modules/vogels/lib/expressions.js:70:44
    at /usr/src/app/node_modules/vogels/node_modules/lodash/index.js:2523:13
    at /usr/src/app/node_modules/vogels/node_modules/lodash/index.js:3073:15
    at baseForOwn (/usr/src/app/node_modules/vogels/node_modules/lodash/index.js:2046:14)
    at /usr/src/app/node_modules/vogels/node_modules/lodash/index.js:3043:18
    at baseReduce (/usr/src/app/node_modules/vogels/node_modules/lodash/index.js:2520:7)
    at Function.<anonymous> (/usr/src/app/node_modules/vogels/node_modules/lodash/index.js:3446:13)
    at Object.exports.serializeUpdateExpression (/usr/src/app/node_modules/vogels/lib/expressions.js:53:18)
npm info lifecycle screwdriver-api@3.0.0~start: Failed to exec start script
npm ERR! Linux 3.16.0-4-amd64
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
npm ERR! node v6.3.1
npm ERR! npm  v3.10.3
npm ERR! code ELIFECYCLE
npm ERR! screwdriver-api@3.0.0 start: `./bin/server`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the screwdriver-api@3.0.0 start script './bin/server'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the screwdriver-api package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     ./bin/server
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs screwdriver-api
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls screwdriver-api
npm ERR! There is likely additional logging output above.
npm ERR! Please include the following file with any support request:
npm ERR!     /usr/src/app/npm-debug.log

--- a/index.js
+++ b/index.js
@@ -1,5 +1,6 @@
 'use strict';
 const Datastore = require('screwdriver-datastore-base');
+const nodeify = require('promise-nodeify');
 const schemas = require('screwdriver-data-schema');
 const Bobby = require('screwdriver-dynamic-dynamodb');
 const DEFAULT_REGION = 'us-west-2';
@@ -128,17 +129,21 @@ class Dynamodb extends Datastore {
 
         userData.id = id;
 
-        return client.update(userData, updateOptions, (err, data) => {
-            if (err) {
-                if (err.statusCode === 400) {
-                    return callback(null, null);
+        const clientUpdate = new Promise((resolve, reject) => {
+            client.update(userData, updateOptions, (err, data) => {
+                if (err) {
+                    if (err.statusCode === 400) {
+                        return resolve(null);
+                    }
+
+                    return reject(err);
                 }
 
-                return callback(err);
-            }
-
-            return callback(err, data.toJSON());
+                return resolve(data.toJSON());
+            });
         });
+
+        return nodeify(clientUpdate, callback);
     }
 
     /**

--- a/package.json
+++ b/package.json
@@ -38,8 +38,9 @@
     "sinon": "^1.17.4"
   },
   "dependencies": {
+    "promise-nodeify": "0.1.0",
     "screwdriver-data-schema": "^5.0.4",
     "screwdriver-datastore-base": "^1.0.0",
-    "screwdriver-dynamic-dynamodb": "^2.0.0"
+    "screwdriver-dynamic-dynamodb": "^2.0.4"
   }
 }

--- a/test/index.test.js
+++ b/test/index.test.js
@@ -372,6 +372,22 @@ describe('index test', () => {
                 done();
             });
         });
+
+        it('fails when it encounters a synchronous error', (done) => {
+            const testError = new Error('testError');
+
+            clientMock.update.throws(testError);
+            datastore._update({
+                table: 'pipelines',
+                params: {
+                    id: 'doNotCare',
+                    data: {}
+                }
+            }, (err) => {
+                assert.deepEqual(err, testError);
+                done();
+            });
+        });
     });
 
     describe('scan', () => {

